libraries/chain/include/eosio/chain/backing_store/kv_context_rocksdb.hpp

```diff
@@ -189,23 +190,31 @@ namespace eosio { namespace chain {
       }
    }; // kv_iterator_rocksdb
 
+   template<typename View, typename Iterator, typename Write_session, typename Resource_manager>
```

Comment:
Do we expect that `View` and `Iterator` can really be independent types?  It seems that in both our runtime and testing systems the `Iterator` tightly coupled to (and a inner class of) `View`. 

If that will always be the case, we should remove the degree of freedom for `Iterator` here and using something like:
```
using iterator_type = typename View::iterator_type;
```

That way the `View` will dictate the iterator type.

Reply:
Thank you Bart so much for taking your time to review the code. Yes, the View and Iterator are tightly coupled. I will collapse them.